core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -284,12 +289,18 @@ class KafkaController(val config : KafkaConfig, zkUtils: ZkUtils, val brokerStat
                   }
                   // If the broker is a follower, updates the isr in ZK and notifies the current leader
                   replicaStateMachine.handleStateChanges(Set(PartitionAndReplica(topicAndPartition.topic,
-                    topicAndPartition.partition, id)), OfflineReplica)
+                    topicAndPartition.partition, id)), OfflineReplica, newBatch = false)
                 }
               }
             }
           }
       }
+
+      inLock(controllerContext.controllerLock) {
```

Comment:
@junrao @ijuma, Joel and I discussed this issue. One possible solution is to call `sendRequestsToBrokers()` inside `newBatch()` if any map is not empty, instead of throwing `IllegalStateException`. I think this is OK because as of current implementation (without this patch) `newBatch()` will never throw `IllegalStateException`. And all calls to `newBatch()` and `sendRequestsToBrokers()` are protected by `controllerContext.controllerLock`. Any suggestion?